.github/workflows/ci.yaml

```diff
@@ -0,0 +1,29 @@
+name: Java CI with Gradle
+
+on:
+  push:
+    branches: ["main"]
+  pull_request:
+    branches: ["main"]
+
+permissions:
+  contents: read
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up JDK 17
+        uses: actions/setup-java@v3
+        with:
+          java-version: "17"
+          distribution: "temurin"
+      - name: Build with Gradle
+        uses: gradle/gradle-build-action@v2
+        with:
+          gradle-version: "8.1.1"
+          arguments: build
+      - name: Run Checkstyle with filter
+        uses: iwata-n/actions-checkstyle@v1.0.0
```

Comment:
since we took continuous deployment out of the requirements for this ticket, wondering if we need to make a new ticket for you to set up your remote server or if you want to include that on this one & demo it for tomorrow?

Reply:
setting up my remote server sounds like a good thing to add to this ticket to me. Also i think it would help me better understand how this is all working :)

Reply:
Let's defer this great idea until tomorrow's meeting so we can discuss on Demo what the best solution is. Thank you for that input!